crates/sp-lightclient/src/lib.rs

```diff
@@ -198,11 +207,40 @@ fn verify_slot<Header: HeaderT>(
     pre_digest: &PreDigest<FarmerPublicKey, FarmerPublicKey>,
 ) -> Result<(), ImportError<HashOf<Header>>> {
     let parent_pre_digest =
-        find_pre_digest(parent_header).map_err(|_| ImportError::InvaildPreDigest)?;
+        find_pre_digest(parent_header).map_err(|_| ImportError::InvalidPreDigest)?;
 
     if pre_digest.slot <= parent_pre_digest.slot {
         return Err(ImportError::InvalidSlot);
     }
 
     Ok(())
 }
+
+fn verify_block_signature<Header: HeaderT>(
+    header: &Header,
+    public_key: &FarmerPublicKey,
+) -> Result<(), ImportError<HashOf<Header>>> {
+    let mut header = header.clone();
```

Comment:
If you need to own a header, you need to have `mut header: Header` in the API. Caller might already be able to give you a copy, but such API forces unnecessary cloning here.